plugins/gateway/tyk/v3.2.2/plugin.go

```diff
@@ -0,0 +1,257 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/TykTechnologies/tyk/ctx"
+	"github.com/TykTechnologies/tyk/log"
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+
+	"github.com/apiclarity/apiclarity/plugins/api/client/client"
+	"github.com/apiclarity/apiclarity/plugins/api/client/client/operations"
+	"github.com/apiclarity/apiclarity/plugins/api/client/models"
+)
+
+const (
+	MaxBodySize              = 1000 * 1000
+	RequestIDHeaderKey       = "X-Request-Id"
+	MinimumSeparatedHostSize = 2
+)
+
+var logger = log.Get()
+
+var (
+	host             string
+	gatewayNamespace string
+)
+
+//nolint:gochecknoinits
+func init() {
+	host = os.Getenv("APICLARITY_HOST")
+	gatewayNamespace = os.Getenv("TYK_GATEWAY_NAMESPACE")
+}
+
+// Called during post phase for setting the apiDefinition since we dont get it in the response phase.
+//nolint:deadcode
+func PostGetAPIDefinition(rw http.ResponseWriter, r *http.Request) {
+	apiDefinition := ctx.GetDefinition(r)
+	if apiDefinition == nil {
+		apiDefinition = apiDefinitionRetriever(r.Context())
+	}
+
+	if apiDefinition == nil {
+		logger.Error("Failed to get api definition")
+		return
+	}
+	ctx.SetDefinition(r, apiDefinition)
+}
+
+// Called during response phase.
+//nolint:deadcode
+func ResponseSendTelemetry(rw http.ResponseWriter, res *http.Response, req *http.Request) {
+	logger.Info("Handling telemetry")
+	telemetry, err := createTelemetry(res, req)
+	if err != nil {
+		logger.Errorf("Failed to create telemetry: %v", err)
+		return
+	}
+
+	apiClient := newAPIClient(host)
+	params := operations.NewPostTelemetryParams().WithBody(telemetry)
+
+	_, err = apiClient.Operations.PostTelemetry(params)
+	if err != nil {
+		logger.Errorf("Failed to post telemetry: %v", err)
+		return
+	}
+	logger.Infof("Telemetry has been sent")
+}
+
+func createTelemetry(res *http.Response, req *http.Request) (*models.Telemetry, error) {
+	truncatedBodyReq := false
+	truncatedBodyRes := false
+
+	apiDefinition := ctx.GetDefinition(req)
+	if apiDefinition == nil {
+		return nil, fmt.Errorf("failed to get api definition")
+	}
+
+	host, port := getHostAndPortFromTargetURL(apiDefinition.Proxy.TargetURL)
+	destinationNamespace := getDestinationNamespaceFromHost(host)
+
+	reqBody, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read request body: %v", err)
+	}
+	if len(reqBody) > MaxBodySize {
+		truncatedBodyReq = true
+		reqBody = []byte{}
+	}
```

Comment:
Need to add this support also in Kong

Reply:
There is a task for this https://portshift.atlassian.net/browse/MAN-8027